File: _c++/symulacja_organizmow/Organizmy/Rosliny/Mlecz.cpp
//
//  Mlecz.cpp
//  WirtualnySwiat
//
//  Created by Michal Tarnacki on 10/04/2022.
//

#include "Mlecz.hpp"

Mlecz::Mlecz(Swiat &swiat):Roslina(swiat){

}
void Mlecz::Akcja(){
    for(int i=0; i<3;i++)
        Roslina::Akcja();
}
char Mlecz::Rysowanie() const{
    return 'm';
}
Organizm* Mlecz::Urodz(){
    return new Mlecz(this->swiat);
}
int Mlecz::GetPrawdopodobienstwo(){
    return 10;
}

bool Mlecz::MozeBycPartnerem(Organizm& organizm){
    if(dynamic_cast<Mlecz*>(&organizm))
       return true;
    else
       return false;
}

File: _java/symulacja_organizmow/src/Symulacja/Rosliny/Mlecz.java
package Symulacja.Rosliny;
import Symulacja.*;

import java.awt.*;

public class Mlecz extends Roslina {
    public Mlecz(Swiat swiat){
        super(swiat);
    };
    public char Rysowanie(){
        return 'm';
    };
    public Organizm Urodz(){
        return new Mlecz(this.swiat);
    };
    public int GetPrawdopodobienstwo(){
        return 10;
    };
    public boolean MozeBycPartnerem(Organizm organizm){
        if(organizm instanceof Mlecz)
        return true;
    else
        return false;
    };
    public void Akcja(){
        for(int i=0; i<3;i++)
            super.Akcja();
    };
    @Override
    public Color GetColor(){
        return new Color(0,0,255);
    };
}
